cluster/replication/consumer.go

```diff
@@ -666,3 +617,70 @@ func (c *CopyOpConsumer) processCancelledOp(ctx context.Context, op ShardReplica
 	}
 	return DELETED, nil
 }
+
+// waitForAsyncReplication waits for async replication to complete by checking the status of the async
+// replication every `asyncStatusInterval` seconds.
+// It returns an error if the async replication does not complete within `asyncStatusRetries` attempts.
+// It returns nil if the async replication has completed.
+func (c *CopyOpConsumer) waitForAsyncReplication(
```

Comment:
I'm very open to changing any of the new retry/max errors logic/numbers here, this is just what I'm imagining as a good implementation based on what we've seen so far, please take a look at the logic to make sure I'm not missing anything obvious as well